File: chatbot/src/main/java/com/chatbot/chatbot/model/ChatMessage.java
package com.chatbot.chatbot.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import com.fasterxml.jackson.annotation.JsonIgnore;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

@Entity
@Table(name = "chat_messages")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ChatMessage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String sender; // "user" or "bot"

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    private String timestamp;
    
    private String imageUrl; // For storing uploaded image URLs
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    @JsonIgnore
    private User user;

    public ChatMessage(String sender, String content) {
        this.sender = sender;
        this.content = content;
        this.timestamp = LocalDateTime.now().format(DateTimeFormatter.ofPattern("HH:mm"));
    }
}

File: aura_bot_flutter/lib/domain/entities/chat_message.dart
import 'package:aura_bot_flutter/domain/entities/detected_object.dart';

class ChatMessage {
  final String id;
  final String sender; // 'user' or 'bot'
  final String content;
  final DateTime timestamp;
  final String? imageUrl;
  final String? ocrText;
  final List<DetectedObject>? detectedObjects;

  const ChatMessage({
    required this.id,
    required this.sender,
    required this.content,
    required this.timestamp,
    this.imageUrl,
    this.ocrText,
    this.detectedObjects,
  });
}
